### PR TITLE
tests: add tests for certs.go and trie.go

### PR DESCRIPTION
**Pull request by brendanashworth**
Just adds some really basic tests. It doesn't test everything in both files, but its a start.

### CODE REVIEW

**Comment by brendanashworth** [comment]
Windows doesn't like this 😢 will fix.

File: certs_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+package main
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestGetCertsPath(t *testing.T) {
```
Comment (harshavardhana) [inline certs_test.go:27]: Comments are missing.

File: certs_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	if path == "" {
+		t.Fail()
+	}
+	if !strings.ContainsRune(path, '/') {
+		t.Fail()
+	}
+
+	mustGetCertsPath()
+}
+
```
Comment (harshavardhana) [inline certs_test.go:47]: Comment..

File: trie_test.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"testing"
+)
+
```
Comment (harshavardhana) [inline trie_test.go:22]: Write comments for each.

File: certs_test.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import (
+	"strings"
+	"testing"
+)
+
+func TestGetCertsPath(t *testing.T) {
+	path, err := getCertsPath()
+	if err != nil {
+		t.Fail()
```
Comment (harshavardhana) [inline certs_test.go]: Make sure to fail with a proper message.

Comment (harshavardhana) [inline certs_test.go]: Same everywhere.

File: certs_test.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	}
+	if certsPath == "" {
+		t.Errorf("expected path to not be an empty string, got: '%s'", certsPath)
+	}
+	// Ensure it contains some sort of path separator.
+	if !strings.ContainsRune(certsPath, os.PathSeparator) {
+		t.Errorf("expected path to contain file separator")
+	}
+	// It should also be an absolute path.
+	if !path.IsAbs(certsPath) {
```
Comment (harshavardhana) [inline certs_test.go]: you have to use 'filepath.IsAbs()' since 'getCertsPath' returns a windows specific path.

Comment (brendanashworth) [inline certs_test.go]: I'm just guessing now because I don't have any windows machines :) thanks for the tip, I'll fix that.

**Pull request merged**